src/network.test.js

```diff
@@ -0,0 +1,110 @@
+import { NetworkStatuses, Monitor, Monitors } from './constants'
+
+import Network from './index'
+
+describe('Network', () => {
+    let monitor
+
+    beforeEach(() => {
+        monitor = new Network()
+        monitor.eventEmitter.addEventListener = jest.fn()
+        monitor.monitors[Monitor.SERVICE].handleError = jest.fn()
+    })
+
+    afterEach(() => {
+        jest.clearAllMocks()
+    })
+
+    describe('constructor', () => {
+        it('initializes with the correct props', () => {
+            expect(monitor).toMatchSnapshot()
+        })
+    })
+
+    describe('on', () => {
+        it('throws error on bad input', () => {
+            expect(() => {
+                monitor.on({}, jest.fn)
+            }).toThrow(Error)
+        })
+        it('throws error on string not in NetworkStatuses', () => {
+            expect(() => {
+                monitor.on('INVALID', jest.fn)
+            }).toThrow(Error)
+        })
+        it('adds an EventListener to the monitor eventEmitter', () => {
+            monitor.on('online', jest.fn)
+            expect(monitor.eventEmitter.addEventListener).toHaveBeenCalledTimes(
+                1,
+            )
+        })
+    })
+
+    describe('all', () => {
+        it('Adds an EventListener for each NetworkStatus', () => {
+            monitor.all(jest.fn)
+            expect(monitor.eventEmitter.addEventListener).toHaveBeenCalledTimes(
+                NetworkStatuses.length,
+            )
+        })
+    })
+
+    describe('serviceError', () => {
+        it('Calls handleError from the service monitor', () => {
+            monitor.serviceError('', {})
```

Comment:
We should be passing in expected arg types here

```suggestion
            monitor.serviceError('/api/resource/1', 502)
```